### PR TITLE
Add karma webworker test

### PR DESCRIPTION
**Pull request by WenheLI**
Sorry for the huge commits history, this is because the branch is out of date.

For this PR, I added support of Karam webworker test, for now, one can test webworker env by using  `yarn test-webworker`

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1833)

### CODE REVIEW

File: src/tests.ts [hunk 1]
```diff
@@ -107,3 +107,4 @@ import './types_test';
 import './util_test';
 import './variable_test';
 import './version_test';
+import './worker/worker_test';
```
Comment (Lewuathe) [inline src/tests.ts]: This error may be caused by the invalid import path here?
```
Step #1 - "build": Error: Looks like you added a new test file. Please run `yarn build` to regenerate src/tests.ts
```

```suggestion
import './worker_test';
```

Comment (WenheLI) [reply src/tests.ts]: Great! Thanks for pointing out that!

**Comment by WenheLI** [comment]
@Lewuathe - Opoos, it seems that I need your confirm for this commits? Would you mind doing so? Or I can revert this commit. 
Thanks!

**Comment by Lewuathe** [comment]
@WenheLI I think we can ignore it because I have already signed CLA. It should not be a problem, I think.

**Comment by Lewuathe** [comment]
But another failure is happening in CI?

```
Failures:
1) computation in worker test-async-cpu undefined tensor in worker
Message:
ReferenceError: Worker is not defined
Stack:
at <Jasmine>
at UserContext.<anonymous> (/workspace/dist/worker_test.js:21:22)
at <Jasmine>
at runCallback (timers.js:705:18)
at tryOnImmediate (timers.js:676:5)
at processImmediate (timers.js:658:5)
```

**Comment by WenheLI** [comment]
Sure, that's probably due to the node running env.

**Comment by Lewuathe** [comment]
Could you also resolve the lint issue?

```
$ tslint -p . -t verbose
ERROR: (prefer-const) /workspace/src/worker_test.ts[5, 5]: Identifier 'workerTest' is never reassigned; use 'const' instead of 'let'.
ERROR: (semicolon) /workspace/src/worker_test.ts[7, 64]: Missing semicolon
ERROR: (prefer-const) /workspace/src/worker_test.ts[9, 6]: Identifier 'b' is never reassigned; use 'const' instead of 'let'.
ERROR: (semicolon) /workspace/src/worker_test.ts[13, 2]: Missing semicolon
ERROR: (prefer-const) /workspace/src/worker_test.ts[15, 5]: Identifier 'fn2workerURL' is never reassigned; use 'const' instead of 'let'.
ERROR: (no-var-keyword) /workspace/src/worker_test.ts[16, 2]: Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: (prefer-const) /workspace/src/worker_test.ts[16, 6]: Identifier 'blob' is never reassigned; use 'const' instead of 'var'.
ERROR: (semicolon) /workspace/src/worker_test.ts[19, 2]: Missing semicolon
```

**Comment by WenheLI** [comment]
@Lewuathe Fixed! Sorry for lint checking before push!

**Pull request closed without merging**